src/sentry/static/sentry/app/components/dataPrivacyRules/dialog.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import Modal from 'react-bootstrap/lib/Modal';
+import omit from 'lodash/omit';
+
+import {Client} from 'app/api';
+import Button from 'app/components/button';
+import ButtonBar from 'app/components/buttonBar';
+import {t} from 'app/locale';
+import {defined} from 'app/utils';
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
+
+import Form from './form';
+import {submitRule} from './submitRule';
+import {handleError} from './handleError';
+import {
+  RuleType,
+  MethodType,
+  Rule,
+  SourceSuggestion,
+  EventId,
+  Errors,
+  RequestError,
+} from './types';
+
+const DEFAULT_RULE_SOURCE_VALUE = '';
+
+type Props = {
+  api: Client;
+  endpoint: string;
+  onClose: () => void;
+  onSubmitSuccess?: (rule: Rule) => void;
+  onUpdateEventId?: (eventId: string) => void;
+  sourceSuggestions?: Array<SourceSuggestion>;
+  eventId?: EventId;
+  rule?: Rule;
+  disabled?: boolean;
+};
+
+type State = {
+  rule: Rule;
+  isFormValid: boolean;
+  errors: Errors;
+};
+
+class Dialog extends React.Component<Props, State> {
+  state: State = {
+    rule: {
+      id: defined(this.props.rule?.id) ? this.props.rule?.id! : -1,
```

Comment:
What does an id of -1 signify? Should you use an absent value like undefined instead?